### PR TITLE
Cypress/e2e tests

### PR DESCRIPTION
**Pull request by madhu-355**

### CODE REVIEW

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10227639105). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

File: .github/workflows/e2e-cypress-tests.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+name: End to End Cypress Test
+
+on:
+  push:
+    branches:
+      - cypress/e2e-tests
```
Comment (shahharsh176) [inline .github/workflows/e2e-cypress-tests.yml:6]: change branch to main

File: .github/workflows/e2e-cypress-tests.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    # paths:
```
Comment (shahharsh176) [inline .github/workflows/e2e-cypress-tests.yml:7]: paths will be 

- frontend
- e2e-tests

Comment (shahharsh176) [reply .github/workflows/e2e-cypress-tests.yml:7]:
```
workflow_run:
    workflows: [Push build to Main]
    types:
      - completed
 ```

File: frontend/package.json [hunk 3]
```diff
@@ -36,7 +36,7 @@
 		"react-scripts": "5.0.1",
 		"react-window": "^1.8.8",
 		"shortid": "^2.2.16",
-		"uglifyjs-webpack-plugin": "^2.2.0",
+		"uglifyjs-webpack-plugin": "^1.1.2",
```
Comment (shahharsh176) [inline frontend/package.json:39]: revert this change

File: frontend/yarn.lock [hunk 4]
```diff
@@ -2743,11 +2743,6 @@ airbnb-prop-types@^2.14.0, airbnb-prop-types@^2.15.0, airbnb-prop-types@^2.16.0:
     prop-types-exact "^1.2.0"
     react-is "^16.13.1"
 
-ajv-errors@^1.0.0:
```
Comment (shahharsh176) [inline frontend/yarn.lock:2746]: revert changes of this file

**Review by shahharsh176 — CHANGES_REQUESTED** [review]

File: e2e-tests/cucumber-html-report.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+const report = require("multiple-cucumber-html-reporter");
+
+report.generate({
+  jsonDir: "./cypress/cucumber-json",
+  reportPath: "./reports/cucumber-html-report.html",
+  metadata: {
```
Comment (kc-diabeat) [inline e2e-tests/cucumber-html-report.js:6]: @madhu-355 I think metadata (e.g. browser device, platform) should be set from the environment and not hardcoded.

File: e2e-tests/cucumber-html-report.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    device: "MacBook Air",
+    platform: {
+      name: "Mac OS",
+      version: "Sonoma 14.2",
+    },
+  },
+  customData: {
+    title: "Run info",
+    data: [
+      { label: "Project", value: "Zango e2e-Tests Report" },
```
Comment (kc-diabeat) [inline e2e-tests/cucumber-html-report.js:20]: We should explore the possibility of showing Zango version as well @madhu-355

File: frontend/src/metadata.json [hunk 7]
```diff
@@ -1 +1 @@
-{"buildMajor":0,"buildMinor":3,"buildPatch":0,"buildTag":""}
\ No newline at end of file
+{"buildMajor":2,"buildMinor":0,"buildPatch":7,"buildTag":""}
```
Comment (kc-diabeat) [inline frontend/src/metadata.json:1]: whats the rationale behind this version jump? @madhu-355 @shahharsh176

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10228771383). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10228784835). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10234288863). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10234564660). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10234914845). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10242919389). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10242939023). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10243014035). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10243035293). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10243223885). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10263581170). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10264190945). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10264270946). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10264298863). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10264400013). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10264423776). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10264464166). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10264497348). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10264636911). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Pull request closed without merging**